Remotion/Web/Development.WebTesting/ScreenshotCreation/BrowserContentLocators/FirefoxBrowserContentLocator.cs

```diff
@@ -60,13 +62,36 @@ public Rectangle GetBrowserContentBounds (IWebDriver driver)
       return result;
     }
 
+    [NotNull]
     private static AutomationElement GetFirefoxContentElement (AutomationElement firefoxWindow)
+    {
+      foreach (var _ in Enumerable.Range (0, 3))
+      {
+        try
+        {
+          var result = GetFirefoxDocumentControl (firefoxWindow);
+
+          if (result != null)
+            return result;
+        }
+        catch (ElementNotAvailableException)
+        {
+        }
+
+        Thread.Sleep (TimeSpan.FromSeconds (1));
+      }
+
+      throw new TimeoutException ("The Firefox content window could not be found within a timeout of 2 seconds.");
```

Comment:
Actually, after looking at this again, I believe the main selling point is the retry logic and not the timeout logic. I'm actually more in favor of an `InvalidOperationException` that goes like "...could not be found after 3 retries." or something like that. Thoughts?

Reply:
Just to make this clear: To me, it doesn't feel like a timeout, but like a retry that happens to have a timeout and was wondering what your thoughts are on that

Reply:
Since the calling method in this class has no way to configure either the timeout or the retry count, the only measurable effect is the timeout, thus the `TimeoutException`. Considering the use in the web tests, the method is used to get coordinates for mostly screenshot operations - the tester using this functionality might be better off seeing the `InvalidOperationException`.

So you are right, this probably should be an `InvalidOperationException`.

Interested in @MichaelKetting 's opinion.

Reply:
TimeoutException.

Reply:
Erm, just read the first comment. Yes, we don't find the window within an acceptable timeframe (=Timeout) but instead, we don't find the window and just stop when it's appropriate (=invalidopex)